src/Microsoft.PowerShell.Commands.Management/commands/management/Service.cs

```diff
@@ -1870,6 +1886,34 @@ protected override void ProcessRecord()
                         }
                     }
 
+                    // Handle the '-SecurityDescriptorSddl' parameter
+                    if(!string.IsNullOrEmpty(SecurityDescriptorSddl))
+                    {
+                        var rawSecurityDescriptor = new RawSecurityDescriptor(SecurityDescriptorSddl);
+                        RawAcl rawDiscretionaryAcl  = rawSecurityDescriptor.DiscretionaryAcl ;
+                        var  discretionaryAcl   = new DiscretionaryAcl (false, false, rawDiscretionaryAcl );
+
+                        byte[] rawDacl = new byte[discretionaryAcl.BinaryLength];
+                        discretionaryAcl.GetBinaryForm(rawDacl, 0);
+                        rawSecurityDescriptor.DiscretionaryAcl = new RawAcl(rawDacl, 0);
+                        byte[] securityDescriptorByte = new byte[rawSecurityDescriptor.BinaryLength];
```

Comment:
For safety, I think we should pass a pointer from a pinned handle for any data we pass to native APIs.  
```
hDacl = GCHandle.Alloc(rawDacl, GCHandleType.Pinned);
lpDacl = hDaclAddrOfPinnedObject();
...
hDacl.Value.Free();
```

Reply:
Never used pointers, will understand about the pinning handle and update. Thanks.

Reply:
When I use the above code and pass  the lpDacl to
`SetServiceObjectSecurity(
hService,
SecurityInfos.DiscretionaryAcl,
lpDacl)` method, the build passes, but the console crashes when the cmdlet is executed for this use case.

Reply:
@PaulHigin what's the purpose of pinning the handle and then immediately passing and freeing it?

Reply:
Just to ensure the GC doesn't move data while native apps access it.

Reply:
@PaulHigin , with the help of @IISResetMe I have made the changes as per the comments, requesting your review.

Reply:
@PaulHigin if it's to enforce common usage throughout the code base, then fair enough, but it is wholly unnecessary in this case - (non-nested one-dimensional arrays of blittable types are implicitly pinned for the duration of the marshalled transaction)[https://github.com/dotnet/corefx/blob/103ef55a43b93815277a0b6771e19675730e8ab7/Documentation/coding-guidelines/interop-pinvokes.md#blittable-types], so passing `byte[]` without explicitly wrapping it in a pinned handle would be just as safe

Reply:
I think @PaulHigin request _explicit_ pin that is more safe.

Reply:
@IISResetMe  I wasn't aware that blittable types are now implicitly pinned ... I was only advocating the pinning of blittable types:

`An instance with nonprimitive (non-blittable) members cannot be pinned. `

@kvprasoon , it looks like this explicit handle pinning is no longer necessary.  Please remove it.  Sorry for the churn!

Reply:
@IISResetMe  Thanks for the correction!

Reply:
You're most welcome @PaulHigin! 

It did get me thinking though, does it make sense to provide a common wrapper for the pinned handle pattern for use in all pinvoke cases where we pass non-blittable types though? Something like this https://gist.github.com/IISResetMe/1b428db41658cef50f51bc168aee3cc1

Reply:
I like the wrapper ... very cool and useful.